src/appleseed/renderer/kernel/shading/shadingpoint.cpp

```diff
@@ -967,13 +967,60 @@ void ShadingPoint::compute_alpha() const
 
       case PrimitiveCurve1:
       case PrimitiveCurve3:
-        // todo: interpolate per vertex alpha for curves here.
+        {
+            assert(is_curve_primitive());
+
+            const GScalar v = m_bary[1];
+
+            if (const CurveObject* curves = static_cast<const CurveObject*>(&get_object()))
+            {
+                const Alpha a = m_primitive_type == PrimitiveCurve1 ?
```

Comment:
The way we indent such constructs:

    const Alpha a =
        m_primitive_type == PrimitiveCurve1
            ? Alpha(curves->get_curve1(m_primitive_index).evaluate_opacity(v))
            : Alpha(curves->get_curve3(m_primitive_index).evaluate_opacity(v));

Reply:
Honestly, since you're already inside a `switch` on the curve type, I would simply get rid of this if and duplicate a tiny bit of code instead.